File: LeetCode/Easy/0020-valid-parentheses/0020-valid-parentheses.cpp
# include <string>
# include <stack>
using namespace std;

class Solution {
public:
    bool isValid(string s) {
        stack<char> stk;
        for (char l : s){
            if (stk.size() == 0){
                stk.push(l);
            }
            else {
                if (l == ')' && stk.top() == '('){
                    stk.pop();
                }
                else if (l == '}' && stk.top() == '{'){
                    stk.pop();
                }
                else if (l == ']' && stk.top() == '['){
                    stk.pop();
                }
                else {
                    stk.push(l);
                }
            }
        }
        return stk.empty();
    }
};

File: LeetCode/Easy/0020-valid-parentheses/0020-valid-parentheses.java
import java.util.ArrayDeque;

class Solution {
    public boolean isValid(String s) {
        char[] list = s.toCharArray();
        ArrayDeque<Character> stk = new ArrayDeque<>();

        for (char c : list){
            if (stk.size() == 0){
                stk.addLast(c);
            }
            else if (c == ')' && stk.peekLast() == '('){
                stk.removeLast();
            }
            else if (c == ']' && stk.peekLast() == '['){
                stk.removeLast();
            }
            else if (c == '}' && stk.peekLast() == '{'){
                stk.removeLast();
            }
            else {
                stk.addLast(c);
            }
        }

        return stk.isEmpty();
    }
}